plugins/producer_plugin/test/test_snapshot_scheduler.cpp

```diff
@@ -118,8 +120,13 @@ BOOST_AUTO_TEST_CASE(snapshot_scheduler_test) {
          // one of the snapshots is done here and request, corresponding to it should be deleted
          BOOST_CHECK_EQUAL(2, pp->get_snapshot_requests().snapshot_requests.size());
 
-         // check whether no pending snapshots present
-         BOOST_CHECK_EQUAL(0, pp->get_snapshot_requests().snapshot_requests.begin()->pending_snapshots->size());
+         // check whether no pending snapshots present for a snapshot with id 0
+         const auto& snapshot_requests = pp->get_snapshot_requests().snapshot_requests;
+         auto it = find_if(snapshot_requests.begin(), snapshot_requests.end(),[](const producer_plugin::snapshot_schedule_information& obj) {return obj.snapshot_request_id == 0;});
+         if (it!=snapshot_requests.end()) {
```

Comment:
At this point, we should know that the snapshot with request ID 0 exists and there is no reason for it to expire.

So instead of an if statement it seems more appropriate to require that it is found before moving on to checking its pending snapshots state.

```cpp
BOOST_REQUIRE(it != snapshot_requests.end());
```

Reply:
Agreed, done.